### ISSUE TITLE
Would it make sense to include some LUKS partition data item to make the response unique?

### ISSUE THREAD

**Issue by cobratbq** [cornelinux/yubikey-luks#70]
This week I have discovered the _yubikey-luks_ package inside debian/ubuntu. I'm reading up a bit on how the mechanism works. Now I'm left with the following question. IIUC the resulting _response_ is - in its basic configuration - solely dependent on the password entered which is used as _challenge_. This means, that if you use this mechanisms on multiple PCs, capturing the result on one PC gives you the passphrase to use on other PC's.

Could we include something unique/identifying of the LUKS partition (e.g. partition UUID) to make the _response_ of the challenge - and therefore the response - distinct for each PC?

__update__ I am having second thoughts about using the UUID, as it is likely changeable, so would corrupt the key slot. I am still curious to your opinion on this.

__update__ to add reasoning:

In the end, you receive a consistent response from the security key, so whatever you do, you cannot protect from capturing the final secret value that unlocks the LUKS partition. This is not a goal.

However, you can avoid that you can discover the secret with __only__ the security key accessible. Including a piece of unique/distinct LUKS/partition information in the challenge, means that you need to have access to the partition at least once - in addition to having access to the security key.

**Comment by Vincent43** [comment]
Yes uuid is changeable so there will be significant risk of locking out user from the data.

**Comment by cobratbq** [comment]
_Updated description to include rationale_: if we use a piece of unique information from the partition/LUKS, we ensure that both the security key and the partition need to be accessible once before the secret value that decrypts the LUKS partition can be discovered.

**Comment by Samdaaman** [comment]
I had a similar idea! 

@cobratbq  check out this [pull request](https://github.com/cornelinux/yubikey-luks/pull/78) (or [this branch](https://github.com/Samdaaman/yubikey-luks/tree/add-hardware-hash-support) on my fork)

Obviously this is a beta feature that I have developed myself so would recommend that you have a master password that you can unlock the partition with normally if the feature fails for whatever reason
